var/spack/repos/builtin/packages/openmpi/nolegacylaunchers.sh

```diff
@@ -1,9 +1,10 @@
 #!/bin/sh
-echo "This version of Spack (openmpi ~legacylaunchers schedulers=slurm) "
-echo "is installed without the mpiexec/mpirun commands to prevent "
-echo "unintended performance issues. See https://github.com/spack/spack/pull/10340 "
-echo "for more details."
-echo "If you understand the potential consequences of a misconfigured mpirun, you can"
-echo "use spack to install 'openmpi+legacylaunchers' to restore the executables."
-echo "Otherwise, use srun to launch your MPI executables."
+cat <<_EOF >&2
+The users are strongly discouraged from using
+'$0'.
+
+Please, refer to your site's user guide for the recommended way to launch MPI
+applications.
```

Comment:
It's more than "strongly discouraged" if the exes in question are completely removed. I'd say "discouraged" should print a warning but "prevented" would be what this script does. How about:

```suggestion
error: '$0' is disabled in this MPI installation.
Please refer to your site's user guide for the recommended way to launch MPI
applications.
```

and add (either in text or a comment for the scrirpt) the link to the relevant spack issues as @alalazo suggests.